### ISSUE TITLE
App gets resetted on Xamarin.UWP after succesfully logging in..

### ISSUE THREAD

**Issue by ExtremeSwat** [xamarin/Xamarin.Auth#202]
# Xamarin.Auth Issue
- [ ] nuget version = 1.5.0.3

### Steps to reproduce

1.Successfully login by using OAuth2 with a provider (such as Google)
2. Handle the Completed event

Platform: UWP
-	[] .NET version = 10.0 - Build 15063

### Expected behaviour

The control should be taken back to the caller instead of resetting the app and losing any info, right ?

### Actual behavior

After the Completed event finishes (or if I have an await within it) the application gets restarted as in the App class (Xamarin Forms) gets instantiated, reinitializing the components and the MainPage

This is the point where I handle the Completed event

![panda1](https://user-images.githubusercontent.com/9104188/27732682-1311b94a-5d9b-11e7-9aed-b56df9a69d88.png)

After the previous statement gets finished  the app itself gets resetted...

![panda2](https://user-images.githubusercontent.com/9104188/27732688-1ba4c0d4-5d9b-11e7-801b-134a45600158.png)

What am I not handling ?

I've attached the Main Xamarin Forms project

[XamarinTestApp.zip](https://github.com/xamarin/Xamarin.Auth/files/1114877/XamarinTestApp.zip)

**Comment by ExtremeSwat** [comment]
EDIT 01 I've found out a fix but I do fear that it might have consequences. I did use NavigationCacheMode="Enabled" instead of it's default, 'Cancel'.
This will basically keep a cached version of the instance once I leave it's context and go to another ContentPage/Page/Frame. Because navigation in UWP doesn't happen like in Xamarin, you have to specify frame.Navigate(typeof(targetPage),args) so you'll prolly have to handle the cache for each page (I guess)
What worries me is that the MainPage and by using this setting from UWP and when the LoadApplication(new SharedProj.App()) this setting will be inherited somewhat by any subsequent ContentPages which might cause issues with the app performance ?
Going to do some tests to check my supposition
```

<forms:WindowsPage
    x:Class="AnonymousCheckerApp.UWP.MainPage"
    xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
    xmlns:forms="using:Xamarin.Forms.Platform.UWP"
    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
    xmlns:local="using:AnonymousCheckerApp.UWP"
    xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
    xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
    mc:Ignorable="d"
    Background="{ThemeResource ApplicationPageBackgroundThemeBrush}"
    NavigationCacheMode="Enabled"
    >
```

**Comment by moljac** [comment]
@ExtremeSwat 

To speed things out can you provide me minimal sample for triaging?

Thanks

**Comment by ExtremeSwat** [comment]
[XamarinTestApp.zip](https://github.com/xamarin/Xamarin.Auth/files/1123699/XamarinTestApp.zip)

I've uploaded the main project only. (XForms)
If you need the whole solution, I've uploaded it to OneDrive
https://softbiz1-my.sharepoint.com/personal/edward_butoiu_cogito-erp_ro/_layouts/15/guestaccess.aspx?docid=14a647776ce304dbbaac4991bfb3d46b4&authkey=AWUWhmixLnQ45pwM6_NgJ9g

**Comment by Brosten** [comment]
Just want to bump this a bit. I experience the same problem. Feels like one of the more critical UWP-issues. Please fix.

**Comment by newky2k** [comment]
@ExtremeSwat @Brosten  could you check and see if this is still an issue with the latest version

Regards

**Comment by Brosten** [comment]
@newky2k Yes, this is still an issue that holds me back from using Xamarin Auth on UWP (I'm using it for iOS and Android...)

**Comment by newky2k** [comment]
@Brosten  Thanks for the update i'll ask @moljac  to look into.

**Comment by cabauman** [comment]
Glad I found this issue. Thought I was the only one for a while. The temporary fix of setting NavigationCacheMode="Enabled" is good enough for now (thanks for that). I'll be following this thread for progress updates.
